src/Mechanic/AstSymbolCollector.fs

```diff
@@ -107,68 +132,94 @@ let getTypeDefnFromPath path =
+    | TraverseStep.TypeDefn d -> d.Range
+    | TraverseStep.Expr e -> e.Range
+    | TraverseStep.MatchClause c -> c.Range
+    | TraverseStep.Binding b -> b.RangeOfBindingAndRhs
+    | TraverseStep.MemberDefn m -> m.Range
+let getParentRangeFromPath path =
+    match path |> List.rev with
+    | [] 
+    | [_] -> None
+    | _::(x::_) -> 
+        Some (getStepRange x)
+
+let getBinding path localRange x =
+    match path with
+        | TraverseStep.Expr _ :: _ -> 
+            getParentRangeFromPath path |> Option.map (fun r -> 
+                getBind localRange [x] |> List.map (defToLocal <| Range.mkRange r.FileName x.RangeOfHeadPat.Start r.End)) |> Option.defaultValue []
+        | _ -> getBind localRange [x]
 
 let getDefSymbols (tree: ParsedInput) =
     let mutable xs = []
     
-    let getFieldsFromTypeDefn = getFieldsAndTypesFromTypeDefn >> List.filter (function |TypeSymbol _, _ -> false |_ -> true) >> List.map fst
-    let getBind = getBind >> List.filter (function |TypeSymbol _, _ -> false |_ -> true) >> List.map fst
-    
     let visitor = { new AstVisitorBase<_>() with    
-        //TODO: union fields
         override __.VisitExpr(_, subExprF, defF, e) =
             match e with
             | _ -> defF e
-        override __.VisitBinding(path, defF, x) = 
-            match path with
-            | TraverseStep.Expr _ :: _ -> defF x
-            | _ ->
-                xs <- xs @ (getBind [x] |> List.map (Symbol.map (fun x -> getNamespace path + "." + x))); defF x
+        
+        override __.VisitLetOrUse(path, bindings, range) = 
+            xs <- xs @ (bindings |> List.collect (getBinding path range >> List.map (DefOrUse.map (fun x -> getNamespace path + "." + x))))
+            None
+            
         override __.VisitComponentInfo(path, _) =
-            let fields = path |> getTypeDefnFromPath |> Option.map getFieldsFromTypeDefn |> Option.defaultValue []
+            let fields = path |> getTypeDefnFromPath |> Option.map getSymbolsFromTypeDefn |> Option.defaultValue []
+            let localRange = 
+                path |> getTypeDefnFromPath |> function 
+                    | Some (SynTypeDefn.TypeDefn(_, SynTypeDefnRepr.ObjectModel _, _, range)) -> Some range
+                    | _ -> None
             let symbolCons = 
                 path |> getTypeDefnFromPath |> function 
-                    | Some (SynTypeDefn.TypeDefn(_, SynTypeDefnRepr.ObjectModel _, _, _)) -> Identificator 
+                    | Some (SynTypeDefn.TypeDefn(_, SynTypeDefnRepr.ObjectModel _, _, _)) -> Identificator
                     | _ -> TypeSymbol
-            xs <- xs @ [getNamespace path |> symbolCons] @ (fields |> List.map (Symbol.map (fun s -> (getNamespace path |> Namespace.removeLastPart) + "." + s))); None
+            xs <- xs @ [getNamespace path |> symbolCons |> mkDef] 
+                @ (fields |> List.map ( 
+                    DefOrUse.map (fun s -> (getNamespace path |> Namespace.removeLastPart) + "." + s)  
```

Comment:
Can you please rewrite that part and add couple of intermediate names and remove >> - that will make it easier to understand for others

Reply:
Thanks for feedback. I made changes in my branch, do I need to open new PR?

Reply:
yes since this is already merged ;-)